packages/mrgn-ui/src/components/action-box-v2/actions/deposit-swap-box/deposit-swap-box.tsx

```diff
@@ -300,194 +285,47 @@ export const DepositSwapBox = ({
+    const props: ExecuteDepositSwapActionProps  = {
+      actionTxns,
+      attemptUuid: uuidv4(),
+      marginfiClient,
       processOpts: {
         ...priorityFees,
-        broadcastType: transactionSettings.broadcastType,
+        broadcastType: transactionSettings?.broadcastType,
       },
       txOpts: {},
-      actionType: lendMode,
-      swapBank: selectedSwapBank,
-    } as ExecuteDepositSwapActionProps;
-
-    await executeAction(params, {
-      captureEvent: captureEvent,
-      setIsLoading: setIsTransactionExecuting,
-      handleOnComplete: (txnSigs: string[]) => {
-        setIsActionComplete(true);
-        setPreviousTxn({
-          txn: txnSigs[txnSigs.length - 1] ?? "",
-          txnType: "DEPOSIT_SWAP",
-          depositSwapOptions: {
-            depositBank: selectedDepositBank as ActiveBankInfo,
-            swapBank: selectedSwapBank as ActiveBankInfo,
-            depositAmount: actionTxns?.actionQuote
-              ? Number(
-                  nativeToUi(actionTxns.actionQuote?.outAmount, selectedDepositBank?.info.rawBank.mintDecimals ?? 9)
-                )
-              : debouncedAmount,
-            swapAmount: actionTxns?.actionQuote ? debouncedAmount : 0,
-          },
-        });
-        onComplete &&
-          onComplete({
-            txn: txnSigs[txnSigs.length - 1] ?? "",
-            txnType: "DEPOSIT_SWAP",
-            depositSwapOptions: {
-              depositBank: selectedDepositBank as ActiveBankInfo,
-              swapBank: selectedSwapBank as ActiveBankInfo,
-              depositAmount: actionTxns?.actionQuote
-                ? Number(
-                    nativeToUi(actionTxns.actionQuote?.outAmount, selectedDepositBank?.info.rawBank.mintDecimals ?? 9)
-                  )
-                : debouncedAmount,
-              swapAmount: actionTxns?.actionQuote ? debouncedAmount : 0,
-              walletToken: walletTokens?.find(
-                (token) => token.address.toBase58() === selectedSwapBank?.address.toBase58()
-              ),
-            },
-          });
+      callbacks: {
+        captureEvent: captureEvent ,
+        onComplete: () => {
+          onComplete?.({
+            walletToken: selectedSwapBank &&  "info" in selectedSwapBank ? undefined : selectedSwapBank ?? undefined ,
+          })
+        } ,
       },
-      retryCallback: (txns: DepositSwapActionTxns, multiStepToast: MultiStepToastHandle) => {
-        retryDepositSwapAction({ ...params, actionTxns: txns, multiStepToast }, selectedSwapBank);
+      infoProps: {
+        depositToken: selectedDepositBank?.meta.tokenSymbol ??'',
+        swapToken: selectedSwapBank ? "info" in selectedSwapBank ? selectedSwapBank.meta.tokenSymbol : selectedSwapBank.symbol : "",
+        depositAmount: dynamicNumeralFormatter(Number(actionTxns.actionQuote?.outAmount) ? Number(nativeToUi(Number(actionTxns.actionQuote?.outAmount), selectedDepositBank?.info.rawBank.mintDecimals ?? 9)) :debouncedAmount ?? 0),
```

Comment:
Number(undefined) will error, the rest of the conditional looks a bit weird as well

perhaps also add some description when debouncedAmount would override actionQuote

I think a conditional (fe. isSwapping) a bit higher would make a bit more sense structure wise